h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -183,6 +180,17 @@ public static double l2norm(double [] x, boolean skipLast){
     return res;
   }
 
+  public static <T extends Iced> T[][] deepClone(T [][] ary){
```

Comment:
Yeah, it's not needed.  The assumption is that the arrays are not allocated at the time of calling the makeCopy (makes only shallow clone of DHistogram). 
I'd feel better that if that would change in the future gbm would fail instead of quietly doing the full clone including the histo data.